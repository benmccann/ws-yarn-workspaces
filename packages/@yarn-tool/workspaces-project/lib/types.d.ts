import type { IPackageJson } from '@ts-type/package-dts';
import type { ILernaJsonCommand } from '@ts-type/package-dts/lerna-json';
import type { IPackageJsonDependenciesField, IDependency } from '@ts-type/package-dts/lib/package-json/types';
import type { FileResult, HostedGitResult, URLResult, AliasResult, RegistryResult } from 'npm-package-arg';
import type { ITSOverwrite } from 'ts-type';
import { IReleaseType } from '@ts-type/package-dts/lerna-json';
export declare const enum EnumWorkspacesProject {
    workspace = "packages/*",
    independent = "independent"
}
export interface IWorkspacesProjectInternal {
    defaultWorkspace: string;
    workspaces: string[];
    releaseConfig: ITSOverwrite<ILernaJsonCommand["version"] & ILernaJsonCommand["publish"], {
        changelogPreset: ILernaJsonCommand["version"]['changelogPreset'];
        conventionalGraduate: ILernaJsonCommand["publish"]['conventionalGraduate'];
        bump: IReleaseType;
    }>;
}
export type IResolvedResult = FileResult | HostedGitResult | URLResult | AliasResult | RegistryResult;
export interface IPackedManifestCore extends Record<IPackageJsonDependenciesField, IDependency> {
    name: string;
    location: string;
    private: boolean;
    resolved: IResolvedResult;
    rootPath: string;
    __isLernaPackage: boolean;
    bin: Record<string, string>;
    scripts: IPackageJson["scripts"];
    /**
     * package.json
     */
    manifestLocation: string;
    /**
     * node_modules
     */
    nodeModulesLocation: string;
    /**
     * node_modules/.bin
     */
    binLocation: string;
    version: string;
    contents: string;
    toJSON(): IPackageJson;
    refresh(): Promise<IPackedManifest>;
    serialize(): Promise<IPackedManifest>;
    updateLocalDependency(resolved: IResolvedResult, depVersion: string, savePrefix: string): any;
}
export interface IPackedManifest extends IPackedManifestCore, Omit<IPackageJson, keyof IPackedManifestCore> {
}
