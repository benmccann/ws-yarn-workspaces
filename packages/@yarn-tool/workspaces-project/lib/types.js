"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnumWorkspacesProject = void 0;
var EnumWorkspacesProject;
(function (EnumWorkspacesProject) {
    EnumWorkspacesProject["workspace"] = "packages/*";
    EnumWorkspacesProject["independent"] = "independent";
})(EnumWorkspacesProject || (exports.EnumWorkspacesProject = EnumWorkspacesProject = {}));
//# sourceMappingURL=types.js.map