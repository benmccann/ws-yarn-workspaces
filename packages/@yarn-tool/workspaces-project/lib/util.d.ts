export declare const enum EnumCheckPaths {
    root = 1,
    rootPath = 0,
    failed = -1
}
export declare function checkPaths(input: {
    root: any;
    rootPath: any;
}, options?: {
    skipStrictCheck?: boolean;
}): EnumCheckPaths;
