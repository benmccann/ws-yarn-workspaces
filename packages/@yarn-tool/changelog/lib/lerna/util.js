"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.handleOptions = void 0;
const lodash_1 = require("lodash");
function handleOptions(options) {
    options = lodash_1.defaults(options !== null && options !== void 0 ? options : {}, {
        type: 'independent',
        changelogPreset: '@bluelovers/conventional-changelog-bluelovers',
        tagPrefix: 'v',
    });
    if (!options.changelogPreset) {
        options.changelogPreset = tryRequire('@bluelovers/conventional-changelog-bluelovers') ? '@bluelovers/conventional-changelog-bluelovers' : 'conventional-changelog-angular';
    }
    return options;
}
exports.handleOptions = handleOptions;
function tryRequire(name) {
    try {
        return require.resolve(name);
    }
    catch (err) {
        if (err.code !== "MODULE_NOT_FOUND") {
            throw new err;
        }
    }
}
//# sourceMappingURL=util.js.map