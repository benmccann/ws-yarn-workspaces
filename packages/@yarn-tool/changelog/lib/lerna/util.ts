import {
	IOptionsWithType,
	IOptionsUpdateChangelog,
	IOptionsRecommendVersion,
	IOptionsBaseCore,
	IChangelogPreset, IOptionsBase,
} from '../types';
import { defaults } from 'lodash';
import { requireResolveExtra, requireResolveCore } from '@yarn-tool/require-resolve';
import { join } from 'path';

export function handleOptions<T extends IOptionsRecommendVersion | IOptionsUpdateChangelog>(options?: IOptionsWithType<T>): IOptionsWithType<T>
{
	const defaultChangelogPreset = '@bluelovers/conventional-changelog-bluelovers' as const;

	options = defaults(options ?? {} as null, <IOptionsWithType<IOptionsBase>>{
		type: 'independent',
		changelogPreset: void 0,
		tagPrefix: 'v',
	})

	if (typeof options.type !== 'string' || !options.type.length)
	{
		options.type = 'independent'
	}

	if (typeof options.tagPrefix !== 'string')
	{
		options.tagPrefix = 'v'
	}

	if (typeof options.changelogPreset !== 'string' || !options.changelogPreset.length || !Boolean(options.changelogPreset))
	{
		options.changelogPreset = void 0;
	}

	if (!options.changelogPreset || options.changelogPreset === defaultChangelogPreset)
	{
		options.changelogPreset = requireResolveExtra(defaultChangelogPreset, {
			includeGlobal: true,
			includeCurrentDirectory: true,
			paths: [
				join(__dirname, '../..'),
			],
		}).result ?? options.changelogPreset;
	}

	options.changelogPreset ??= 'conventional-changelog-angular';

	return options
}

function tryRequire(name: IChangelogPreset)
{
	try
	{
		return requireResolveCore(name, {
			includeGlobal: true,
			includeCurrentDirectory: true,
		})
	}
	catch (err)
	{
		if (err.code !== "MODULE_NOT_FOUND")
		{
			throw new err
		}
	}
}
