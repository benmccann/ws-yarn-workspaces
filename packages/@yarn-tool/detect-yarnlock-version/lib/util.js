"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getMetadataVersion = getMetadataVersion;
exports._getMetadataVersionCore = _getMetadataVersionCore;
function getMetadataVersion(yarnLockObject) {
    return _getMetadataVersionCore(yarnLockObject.__metadata);
}
function _getMetadataVersionCore(__metadata) {
    var _a;
    return (_a = __metadata === null || __metadata === void 0 ? void 0 : __metadata.version) === null || _a === void 0 ? void 0 : _a.toString();
}
//# sourceMappingURL=util.js.map