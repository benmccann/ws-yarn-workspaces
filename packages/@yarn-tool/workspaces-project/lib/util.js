"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.checkPaths = exports.EnumCheckPaths = void 0;
const path_dir_normalize_1 = __importDefault(require("path-dir-normalize"));
var EnumCheckPaths;
(function (EnumCheckPaths) {
    EnumCheckPaths[EnumCheckPaths["root"] = 1] = "root";
    EnumCheckPaths[EnumCheckPaths["rootPath"] = 0] = "rootPath";
    EnumCheckPaths[EnumCheckPaths["failed"] = -1] = "failed";
})(EnumCheckPaths = exports.EnumCheckPaths || (exports.EnumCheckPaths = {}));
function checkPaths(input, options) {
    let { root, rootPath } = input;
    if ((root === null || root === void 0 ? void 0 : root.length) && (rootPath === null || rootPath === void 0 ? void 0 : rootPath.length)) {
        root = path_dir_normalize_1.default(root);
        rootPath = path_dir_normalize_1.default(rootPath);
        if (root !== rootPath) {
            if (rootPath.indexOf(root) === 0) {
                return 1 /* root */;
            }
            else if (!(options === null || options === void 0 ? void 0 : options.skipStrictCheck)) {
                return -1 /* failed */;
            }
        }
        return 0 /* rootPath */;
    }
    else if (rootPath === null || rootPath === void 0 ? void 0 : rootPath.length) {
        return 0 /* rootPath */;
    }
    return -1 /* failed */;
}
exports.checkPaths = checkPaths;
//# sourceMappingURL=util.js.map