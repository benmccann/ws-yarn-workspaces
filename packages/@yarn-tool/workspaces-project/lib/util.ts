import pathDirNormalize from 'path-dir-normalize';

export const enum EnumCheckPaths
{
	root = 1,
	rootPath = 0,
	failed = -1
}

export function checkPaths(input: {
	root,
	rootPath,
}, options?: {
	skipStrictCheck?: boolean
}): EnumCheckPaths
{
	let { root, rootPath } = input

	if (root?.length && rootPath?.length)
	{
		root = pathDirNormalize(root)
		rootPath = pathDirNormalize(rootPath)

		if (root !== rootPath)
		{
			if (rootPath.indexOf(root) === 0)
			{
				return EnumCheckPaths.root
			}
			else if (!options?.skipStrictCheck)
			{
				return EnumCheckPaths.failed
			}
		}

		return EnumCheckPaths.rootPath
	}
	else if (rootPath?.length)
	{
		return EnumCheckPaths.rootPath
	}

	return EnumCheckPaths.failed
}

{}

