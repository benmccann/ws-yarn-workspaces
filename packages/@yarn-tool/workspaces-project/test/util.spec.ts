import { checkPaths, EnumCheckPaths } from '../lib/util';
import { join } from 'path';

describe(`describe`, () =>
{
	let root = __dirname;

	test(`EnumCheckPaths.root`, () =>
	{
		let rootPath = join(root, 'xxxx');

		let actual = checkPaths({
			root,
			rootPath,
		});
		let expected = EnumCheckPaths.root;

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`EnumCheckPaths.rootPath`, () =>
	{
		let rootPath = join(root);

		let actual = checkPaths({
			root,
			rootPath,
		});
		let expected = EnumCheckPaths.rootPath;

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`failed`, () =>
	{
		let rootPath = join(root, '..');

		let actual = checkPaths({
			root,
			rootPath,
		});
		let expected = EnumCheckPaths.failed;

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`root is empty return EnumCheckPaths.rootPath`, () =>
	{
		let rootPath = join(root);

		let actual = checkPaths({
			root: '',
			rootPath,
		});
		let expected = EnumCheckPaths.rootPath;

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`rootPath is empty return EnumCheckPaths.failed`, () =>
	{
		let rootPath = '';

		let actual = checkPaths({
			root,
			rootPath,
		});
		let expected = EnumCheckPaths.failed;

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`empty return EnumCheckPaths.failed`, () =>
	{
		let rootPath = '';

		let actual = checkPaths({
			root: '',
			rootPath,
		});
		let expected = EnumCheckPaths.failed;

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

});

describe(`skipStrictCheck`, () =>
{
	let root = __dirname;

	let options = {
		skipStrictCheck: true,
	}

	test(`EnumCheckPaths.root`, () =>
	{
		let rootPath = join(root, 'xxxx');

		let actual = checkPaths({
			root,
			rootPath,
		}, options);
		let expected = EnumCheckPaths.root;

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`EnumCheckPaths.rootPath`, () =>
	{
		let rootPath = join(root);

		let actual = checkPaths({
			root,
			rootPath,
		}, options);
		let expected = EnumCheckPaths.rootPath;

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`failed => rootPath`, () =>
	{
		let rootPath = join(root, '..');

		let actual = checkPaths({
			root,
			rootPath,
		}, options);
		let expected = EnumCheckPaths.rootPath

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`root is empty return EnumCheckPaths.rootPath`, () =>
	{
		let rootPath = join(root);

		let actual = checkPaths({
			root: '',
			rootPath,
		}, options);
		let expected = EnumCheckPaths.rootPath;

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`rootPath is empty return EnumCheckPaths.failed`, () =>
	{
		let rootPath = '';

		let actual = checkPaths({
			root,
			rootPath,
		}, options);
		let expected = EnumCheckPaths.failed;

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

	test(`empty return EnumCheckPaths.failed`, () =>
	{
		let rootPath = '';

		let actual = checkPaths({
			root: '',
			rootPath,
		}, options);
		let expected = EnumCheckPaths.failed;

		expect(actual).toStrictEqual(expected);
		//expect(actual).toBeInstanceOf(Date);
		expect(actual).toMatchSnapshot();

	});

})
