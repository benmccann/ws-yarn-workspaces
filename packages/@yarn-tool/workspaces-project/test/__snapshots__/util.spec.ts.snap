// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`describe EnumCheckPaths.root 1`] = `1`;

exports[`describe EnumCheckPaths.rootPath 1`] = `0`;

exports[`describe empty return EnumCheckPaths.failed 1`] = `-1`;

exports[`describe failed 1`] = `-1`;

exports[`describe root is empty return EnumCheckPaths.rootPath 1`] = `0`;

exports[`describe rootPath is empty return EnumCheckPaths.failed 1`] = `-1`;

exports[`skipStrictCheck EnumCheckPaths.root 1`] = `1`;

exports[`skipStrictCheck EnumCheckPaths.rootPath 1`] = `0`;

exports[`skipStrictCheck empty return EnumCheckPaths.failed 1`] = `-1`;

exports[`skipStrictCheck failed => rootPath 1`] = `0`;

exports[`skipStrictCheck root is empty return EnumCheckPaths.rootPath 1`] = `0`;

exports[`skipStrictCheck rootPath is empty return EnumCheckPaths.failed 1`] = `-1`;
